executor/src/witgen/vm_processor.rs

```diff
@@ -155,6 +157,9 @@ impl<'a, 'b, 'c, T: FieldElement, Q: QueryCallback<T>> VmProcessor<'a, 'b, 'c, T
                         loop_detection_log_level,
                         "Found loop with period {p} starting at row {row_index}"
                     );
+                    record_end_identity(profiling::UNUSED_IDENTITY_ID);
+                    profiling::reset_and_print_profile_summary_identity(self.fixed_data);
+                    record_start_identity(profiling::UNUSED_IDENTITY_ID);
```

Comment:
Unfortunately, this breaks if there is more than one VM:
```
$ RUST_LOG=debug cargo run pil test_data/asm/vm_to_vm.asm -o output -f

...
[00:00:00 (ETA: 00:00:00)] ████████████████████ 100% - Starting...                                                                                                                Finalizing VM: Main Machine
Transposing 256 rows with 19 columns...
Finalizing remaining rows...
Needed to finalize 256 / 256 rows.
Done transposing.
Finalizing VM: Secondary machine 0: main_vm (Vm)
Found loop with period 1 starting at row 100

 == Identities processing profile (7464 events)
thread 'main' panicked at executor/src/witgen/machines/profiling.rs:158:17:
Unexpected recursive call!
```

Reply:
Ok, would you think it's a good fix to add a boolean flag to just output the partial profile summary (and then temporarily closing all open calls) and do not reset?

Reply:
Turns out the error was something else: https://github.com/powdr-labs/powdr/pull/1473/files#r1657169269

I still think it's a bit confusing, and at least we should add a debug log here which machine reached the loop and that all future profiling logs will not include any events before this point. As far as I understand, what happens now in `vm_to_vm.asm` is:
1. Profiling starts
2. Main machine reaches the loop -> profiling printed & reseted
3. `Generator::run()` finishes the loop and returns
4. We call `take_witness_col_values` on the secondary machines, which includes the secondary VM. This invokes the VM processor and immediately finds a loop -> profiling printed & reseted
5. Then, we call `reset_and_print_profile_summary_identity` in `WitnessGenerator::generate()`

Not sure this is what we want! I think it would be cleanest to get 3 profiles:
1. Main machine reaches the loop
2. `Generator::run()` returns
3. All the `take_witness_col_values` have completed (this is probably the least important)